util/mount.fuse.c

```diff
@@ -80,6 +93,116 @@ static char *add_option(const char *opt, char *options)
 	return options;
 }
 
+static int prepare_fuse_fd(const char *mountpoint, const char* subtype,
+			   const char *options)
+{
+	int subtype_len = strlen(subtype) + 9;
+	char* options_copy = xrealloc(NULL, subtype_len);
+	snprintf(options_copy, subtype_len, "subtype=%s", subtype);
+	options_copy = add_option(options, options_copy);
+
+	const char *argv[] = { progname, "-o", options_copy };
+	struct fuse_args args = FUSE_ARGS_INIT(sizeof(argv) / sizeof(argv[0]),
+					       (char**) argv);
+	struct fuse_lowlevel_ops dummy_ops;
+	memset(&dummy_ops, 0, sizeof(dummy_ops));
+	struct fuse_session *session = fuse_session_new(
+			&args, &dummy_ops, sizeof(dummy_ops), NULL);
+	free(options_copy);
+	if (session == NULL || fuse_session_mount(session, mountpoint) == -1) {
+		exit(1);
+	}
+
+	/*
+	 * Duplicate the FUSE file descriptor to obtain a copy that's still
+	 * valid after fuse_session_destroy closes its copy. Note that this
+	 * conveniently also gives us a file descriptor with the CLOEXEC flag
```

Comment:
I'd say "without the CLOEXEC flag set", but at least "with the CLOEXEC flag cleared". I think my brain reads "with the CLOEXEC flag" and assumes the flag is *set*.

Reply:
Fixed locally and will be in next upload.